src/core/models.py

```diff
@@ -1919,3 +1968,641 @@ def log_hijack_ended(sender, hijacker, hijacked, request, **kwargs):
+        :type country: str or core.models.Country
+        :type account: core.models.Account
+        :type frozen_author: submission.models.FrozenAuthor
+        :type preprint_author: repository.models.PreprintAuthor
+        """
+        organization, _created = Organization.get_or_create_without_ror(
+            institution=institution,
+            country=country,
+            account=account,
+            frozen_author=frozen_author,
+            preprint_author=preprint_author,
+        )
+
+        # Create or update the actual affiliation if the associated
+        # account / frozen author / preprint author has been saved already
+        try:
+            affiliation, created = Affiliation.objects.get_or_create(
+                account=account,
+                frozen_author=frozen_author,
+                preprint_author=preprint_author,
+                organization=organization,
+                department=department,
+            )
+            affiliation.is_primary = True
+            affiliation.save()
+        except ValueError:
+            logger.warn('The affiliation could not be created.')
+            affiliation = None
+            created = False
+        return affiliation, created
+
+    @classmethod
+    def naive_set_primary_department(
+        cls,
+        value,
+        account=None,
+        frozen_author=None,
+    ):
+        """
+        Backwards-compatible API for setting department names in isolation.
+        It is better to use Affiliation.get_or_create_without_ror where department
+        is set togther with institution and country.
+        Does not support ORCIDs, RORs or multiple affiliations.
+        """
+        # Create or update an affiliation if the associated
+        # account / frozen author / preprint author has been saved already
+        try:
+            affiliation, _created = Affiliation.objects.get_or_create(
+                account=account,
+                frozen_author=frozen_author,
+                is_primary=True,
+            )
+            affiliation.department = value
+            affiliation.save()
+        except ValueError:
+            logger.warn('The department could not be set.')
+
+    def save(self, *args, **kwargs):
+        self.set_primary_if_first(self)
+        self.keep_is_primary_unique(self)
+        super().save(*args, **kwargs)
+
+
+class Location(models.Model):
+    name = models.CharField(
+        max_length=200,
+        help_text="City or place name",
+        blank=True,
+    )
+    country = models.ForeignKey(
+        Country,
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+    )
+    latitude = models.DecimalField(
+        max_digits=9,
+        decimal_places=6,
+        blank=True,
+        null=True,
+    )
+    longitude = models.DecimalField(
+        max_digits=9,
+        decimal_places=6,
+        blank=True,
+        null=True,
+    )
```

Comment:
Are we making use of these fields? If not, I reckon it is worth holding off on exposing an API that we might need to later make backwards compatible if we end up using [PostGIS](https://postgis.net/), similar to the ongoing [KOMET](https://github.com/openlibhums/janeway/issues/1928) project

Reply:
Yeah, good point. Removed these fields.